osf/models/notable_domain.py

```diff
@@ -40,6 +45,56 @@ def __repr__(self):
     def __str__(self):
         return repr(self)
 
+    @staticmethod
+    def check_resource_for_domains(obj, confirm_spam=False, send_to_moderation=False):
+        fields_to_query = []
+        for field in list(obj.SPAM_CHECK_FIELDS):
+            fields_to_query += [models.Q(**{f'{field}__regex': settings.DOMAIN_REGEX})]
+        query = reduce(operator.or_, fields_to_query)
+        has_domain = obj.__class__.objects.filter(query).filter(id=obj.id).exists()
+
+        if not has_domain:
+            return False
+        elif confirm_spam and NotableDomain.has_spam_domain(obj):
+            obj.confirm_spam(save=True)
+        elif send_to_moderation:
+            NotableDomain.add_domains_to_moderation_queue(obj)
+        obj.save()
+        return True
+
+    @staticmethod
+    def add_domains_to_moderation_queue(obj):
+        fields_to_query = []
+        spam_fields = list(obj.SPAM_CHECK_FIELDS)
+        for field in spam_fields:
+            fields_to_query += [models.Q(**{f'{field}__regex': settings.DOMAIN_REGEX})]
+        query = reduce(operator.or_, fields_to_query)
+        raw_field_content = str(obj.__class__.objects.filter(query).filter(id=obj.id).values(*spam_fields))
```

Comment:
Initially I wanted the code to be isolated from the `SpamMixin`, but obviously it makes more sense seeing as the check only takes place in `do_spam_check`. But since the spam content is always passed to `do_spam_check` it is no longer necessary.